lib/src/css_parser.dart

```diff
@@ -70,6 +115,266 @@ class ExpressionMapping {
+
+  static FontWeight expressionToFontWeight(css.Expression value) {
+    if (value is css.NumberTerm) {
+      switch (value.text) {
+        case "100":
+          return FontWeight.w100;
+        case "200":
+          return FontWeight.w200;
+        case "300":
+          return FontWeight.w300;
+        case "400":
+          return FontWeight.w400;
+        case "500":
+          return FontWeight.w500;
+        case "600":
+          return FontWeight.w600;
+        case "700":
+          return FontWeight.w700;
+        case "800":
+          return FontWeight.w800;
+        case "900":
+          return FontWeight.w900;
+      }
+    } else if (value is css.LiteralTerm) {
+      switch(value.text) {
+        case "bold":
+          return FontWeight.bold;
+        case "bolder":
+          return FontWeight.w900;
+        case "lighter":
+          return FontWeight.w200;
+      }
+      return FontWeight.normal;
+    }
+    return FontWeight.normal;
+  }
+
+  static String expressionToFontFamily(css.Expression value) {
+    if (value is css.LiteralTerm) return value.text;
+    return null;
+  }
+
+  static LineHeight expressionToLineHeight(css.Expression value) {
+    if (value is css.NumberTerm) {
+      return LineHeight.number(double.tryParse(value.text));
+    } else if (value is css.PercentageTerm) {
+      return LineHeight.percent(double.tryParse(value.text));
+    } else if (value is css.EmTerm) {
+      return LineHeight.em(double.tryParse(value.text));
+    } else if (value is css.RemTerm) {
+      return LineHeight.rem(double.tryParse(value.text));
+    } else if (value is css.LengthTerm) {
+      return LineHeight(double.tryParse(value.text.replaceAll(new RegExp(r'\s+(\d+\.\d+)\s+'), '')), "length");
+    }
+    return LineHeight.normal;
+  }
+
+  static TextAlign expressionToTextAlign(css.Expression value) {
+    if (value is css.LiteralTerm) {
+      switch(value.text) {
+        case "center":
+          return TextAlign.center;
+        case "left":
+          return TextAlign.left;
+        case "right":
+          return TextAlign.right;
+        case "justify":
+          return TextAlign.justify;
+        case "end":
+          return TextAlign.end;
+        case "start":
+          return TextAlign.start;
+      }
+    }
+    return TextAlign.start;
+  }
+
+  static TextDecoration expressionToTextDecorationLine(List<css.LiteralTerm> value) {
+    List<TextDecoration> decorationList = [];
+    for (css.LiteralTerm term in value) {
+      switch(term.text) {
+        case "overline":
+          decorationList.add(TextDecoration.overline);
+          break;
+        case "underline":
+          decorationList.add(TextDecoration.underline);
+          break;
+        case "line-through":
+          decorationList.add(TextDecoration.lineThrough);
+          break;
+        default:
+          decorationList.add(TextDecoration.none);
+          break;
+      }
+    }
+    return TextDecoration.combine(decorationList);
```

Comment:
Should `TextDecoration.none` not override all the others. And if so, does it, given this code?

Reply:
It does, it is the default case in the switch, so if the first three cases do not apply, it defaults to adding `TextDecoration.none` into the list.

Reply:
I mean if you write (agreed, perhaps against the standard) `none underline overline` should it not only return `TextDecoration.none`?

Reply:
I didn't think of this case at all, thanks! I've fixed it so if the list contains `TextDecoration.none` at all, it will be replaced with just `[TextDecoration.none]`.